### ISSUE TITLE
[C API] Segfault on iplContextRetain

### ISSUE THREAD

**Issue by Aceeri** [ValveSoftware/steam-audio#190]
Hi, I've been trying to figure out how to integrate steam audio into Bevy (a Rust based game engine), was going well until I ran into a segfault coming from trying to commit the simulation. I'm not entirely sure if I have somehow mis-configured this, I'm probably going to keep trying to tinker with it, however the only information I can really get is that the segfault occurs within `iplContextRetain` after calling `iplSimulatorCommit`. I couldn't get much info past that due to lack of debug symbols unfortunately.

I can provide some example code but I think I am going to try writing this entirely in C to make sure I'm not doing something weird with the interop.

**Comment by Aceeri** [comment]
I also tried 4.0.2 and that seemed to work fine, so I think there may have been a regression in between them.

**Comment by LutzenH** [comment]
I am experiencing the same problem when performing `iplSimulatorCommit()` in version 4.0.3.

`Access violation reading location 0x00000010`

Problem does not exist in version 4.0.2 as @Aceeri has said.

**Comment by lakulish** [comment]
@Aceeri @LutzenH What flags did you specify in your `IPLSimulationSettings` when creating the `IPLSimulator` object? We recently found a regression in 4.0.3 where `iplSimulatorCommit` would crash if the `IPL_SIMULATIONFLAGS_REFLECTIONS` flag was not specified.

**Comment by LutzenH** [comment]
My `IPLSimulationSettings` at the time looked like this:

```
IPLSimulationSettings simulation_settings = {};
simulation_settings.flags = IPL_SIMULATIONFLAGS_DIRECT;
simulation_settings.sceneType = IPL_SCENETYPE_DEFAULT;
simulation_settings.samplingRate = state->audio_settings.samplingRate; // 44100
simulation_settings.frameSize = state->audio_settings.frameSize; // 1024

error_code = iplSimulatorCreate(state->context, &simulation_settings, &state->simulator);
if (error_code != IPL_STATUS_SUCCESS) {
	// Error handling & Cleanup
	// ...
}

iplSimulatorSetScene(state->simulator, state->scene);
iplSimulatorCommit(state->simulator); // Crash would happen here if I recall correctly.
```

I Only used the flag `IPL_SIMULATIONFLAGS_DIRECT` because I was not using (baked)reflections yet. Only `IPLBinauralEffect` and `IPLDirectEffect`. I am sadly unable to test and run the code-base from that time. Maybe @Aceeri can give some more insight.

**Comment by Aceeri** [comment]
@lakulish I can verify that setting the reflection flag fixes the issue, thank you!

https://user-images.githubusercontent.com/5142292/162373252-8d5471d3-5ce6-45c2-88e3-b0855f60ed8e.mp4

**Comment by Aceeri** [comment]
I'll keep this open for the sake of visibility for others actually

**Comment by lakulish** [comment]
We've just released Steam Audio 4.1.0, which includes a fix for this issue: https://github.com/ValveSoftware/steam-audio/releases/tag/v4.1.0.
